src/Compilers/CSharp/Test/Symbol/Symbols/PartialPropertiesTests.cs

```diff
@@ -0,0 +1,467 @@
+            comp.VerifyEmitDiagnostics(
+                // (4,17): error CS9304: Property accessor 'C.P.init' must be implemented because it is declared on the definition part
+                //     partial int P { get => throw null!; }
+                Diagnostic(ErrorCode.ERR_PartialPropertyMissingAccessor, "P").WithArguments("C.P.init").WithLocation(4, 17)
+                );
+        }
+
+        [Fact]
+        public void UnexpectedAccessor_01()
+        {
+            // implementation unexpected setter
+            var source = """
+                partial class C
+                {
+                    partial int P { get; }
+                    partial int P { get => throw null!; set { } }
+                }
+                """;
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (4,17): error CS9305: Property accessor 'C.P.set' does not implement any accessor declared on the definition part
+                //     partial int P { get => throw null!; set { } }
+                Diagnostic(ErrorCode.ERR_PartialPropertyUnexpectedAccessor, "P").WithArguments("C.P.set").WithLocation(4, 17)
+                );
+        }
+
+        [Fact]
+        public void UnexpectedAccessor_02()
+        {
+            // implementation unexpected getter
+            var source = """
+                partial class C
+                {
+                    partial int P { set; }
+                    partial int P { get => throw null!; set { } }
+                }
+                """;
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (4,17): error CS9305: Property accessor 'C.P.get' does not implement any accessor declared on the definition part
+                //     partial int P { get => throw null!; set { } }
+                Diagnostic(ErrorCode.ERR_PartialPropertyUnexpectedAccessor, "P").WithArguments("C.P.get").WithLocation(4, 17)
+                );
+        }
+
+        [Fact]
+        public void UnexpectedAccessor_03()
+        {
+            // implementation unexpected init
+            var source = """
+                partial class C
+                {
+                    partial int P { get; }
+                    partial int P { get => throw null!; init { } }
+                }
+                """;
+            var comp = CreateCompilation([source, IsExternalInitTypeDefinition]);
+            comp.VerifyEmitDiagnostics(
+                // (4,17): error CS9305: Property accessor 'C.P.init' does not implement any accessor declared on the definition part
+                //     partial int P { get => throw null!; init { } }
+                Diagnostic(ErrorCode.ERR_PartialPropertyUnexpectedAccessor, "P").WithArguments("C.P.init").WithLocation(4, 17)
+                );
+        }
+
+        [Fact]
+        public void AccessorKind_01()
```

Comment:
Consider adding the converse test too (definition has `init`, but implementation has `set`)